server/engine/src/main/java/org/eclipse/lsp/cobol/implicitDialects/cics/utility/CICSPerformSPOptionsCheckUtility.java

```diff
@@ -311,5 +313,26 @@ private void checkDumpDuplicates(List<TerminalNode> rules, ErrorSeverity severit
                                 "DUMP");
                 });
     }
+    private void checkAll(List<ParseTree> children) {
+        if (children.isEmpty()) return;
+        boolean isAll = false;
+        boolean isResource = false;
+        for (ParseTree child : children) {
+            if (!TerminalNode.class.isAssignableFrom(child.getClass())) continue;
+            int token = ((TerminalNode) child).getSymbol().getType();
+            if (token == CICSLexer.ALL) isAll = true;
+            else if (token != CICSLexer.STATISTICS && token != CICSLexer.RECORD && token != CICSLexer.RESETNOW)
+                isResource = true;
+
+            if (isAll && isResource) {
+                throwException(
+                        ErrorSeverity.ERROR,
+                        getLocality(child),
+                        "Invalid options provided: ",
+                        child.getText());
+                isResource = false;
+            }
+        }
```

Comment:
I think that just one message for the entire context is enough.
```suggestion
            if (isAll && isResource) break;
        }
        if (isAll && isResource) {
            throwException(
                    ErrorSeverity.ERROR,
                    getLocality(ctx),
                    "Option ALL cannot be combined with individual resource types",
                    "");
        }
```